File: code/94/solution.cpp
#include "../template/bigint.cpp"
#include "../template/china_reminder.cpp"
#include "../template/euler.cpp"
#include "../template/pell.cpp"
#include "../template/prime.cpp"
#include "../template/prime_cnt.cpp"
#include <iostream>/*{{{*/
#include <map>
#include <math.h>
#include <set>
#include <vector>
const long long e2 = 100;
const long long e3 = e2 * 10;
const long long e4 = e3 * 10;
const long long e5 = e4 * 10;
const long long e6 = e4 * e2;
const long long e7 = e6 * 10;
const long long e8 = e7 * 10;
long long mod      = 1e9 + 7;
using namespace std; /*}}}*/

/*add{{{*/
long long base_add(long long a, long long b) {
    a %= mod;
    b %= mod;
    return (a + b) % mod;
}
long long add(long long a, long long b) { return base_add(a, b); }
long long add(initializer_list<long long> args) {
    long long ret = 0;
    for (auto i : args)
        ret = base_add(ret, i);
    return ret;
}
/*}}}*/

/*{{{mul*/
long long base_mul(long long a, long long b) {
    a %= mod;
    b %= mod;
    long long ret = 0;
    while (b) {
        if (b & 1)
            ret = (ret + a) % mod;
        b >>= 1;
        a = (a + a) % mod;
    }
    return ret;
}
long long mul(long long a, long long b) { return base_mul(a, b); }
long long mul(initializer_list<long long> args) {
    long long ret = 1;
    for (auto i : args)
        ret = base_mul(ret, i);
    return ret;
} /*}}}*/

/*{{{qpow*/
long long qpow(long long a, long long b) {
    long long ret = 1;
    while (b) {
        if (b & 1)
            ret = mul({ret, a});
        a = mul({a, a});
        b >>= 1;
    }
    return ret;
} /*}}}*/

int main() {
    long long ans1 = 0;
    long long k = 8, t = 15;
    while (true) {
        long long n = k * 2 + 1;
        long long S = n * 3 - 1;
        if (S > 1e9)
            break;
        if (S > 0) {
            ans1 += S;
            cout << n << ' ' << n << ' ' << n - 1 << endl;
        }
        long long kk = -2 * k - t - 2;
        t            = -3 * k - 2 * t - 2;
        k            = kk;
    }

    long long ans2 = 0;
    k = 2, t = 4;
    while (true) {
        long long n = k * 2 + 1;
        long long S = n * 3 + 1;
        if (S > 1e9)
            break;
        if (S > 0) {
            ans2 += S;
            cout << n << ' ' << n << ' ' << n + 1 << endl;
        }
        long long kk = -2 * k - t - 1;
        t            = -3 * k - 2 * t - 1;
        k            = kk;
    }

    cout << ans1 + ans2 << endl;
    return 0;
}

File: code/94/solution.go
package main /*{{{*/

import (
	"fmt"
	"reflect"
)

const (
	e4  = int64(10000)
	e5  = int64(100000)
	e6  = int64(1000000)
	e7  = int64(10000000)
	e8  = e4 * e4
	mod = int64(1e9 + 7)
)

func add(list ...interface{}) int64 { /*{{{*/
	if len(list) == 0 {
		return 0
	}
	ret := int64(0)
	for _, i := range list {
		buf := int64(0)
		switch reflect.TypeOf(i).String() {
		case "int":
			buf = int64(i.(int))
		case "int64":
			buf = i.(int64)
		}
		ret = (ret + buf) % mod
	}
	return ret
} /*}}}*/
func mul(list ...interface{}) int64 { /*{{{*/
	if len(list) == 0 {
		return 0
	}
	ret := int64(1)
	for _, i := range list {
		buf := int64(0)
		switch reflect.TypeOf(i).String() {
		case "int":
			buf = int64(i.(int))
		case "int64":
			buf = i.(int64)
		}
		buf %= mod
		base := ret
		ret = 0
		for buf > 0 {
			if (buf & 1) == 1 {
				ret = add(ret, base)
			}
			base = add(base, base)
			buf >>= 1
		}
	}
	return ret
} /*}}}*/
func qpow(a, b int64) int64 { /*{{{*/
	ret := int64(1)
	for b > 0 {
		if (b & 1) == 1 {
			ret = mul(ret, a)
		}
		a = mul(a, a)
		b >>= 1
	}
	return ret
} /*}}}*/
/*}}}*/

func main() {
	ans1 := int64(0)
	k, t := int64(8), int64(15)
	for true {
		n := k*2 + 1
		S := n*3 - 1
		if S > 1e9 {
			break
		}
		if S > 0 {
			ans1 += S
			fmt.Println(n, n, n-1)
		}
		k, t = -2*k-t-2, -3*k-2*t-2
	}

	ans2 := int64(0)
	k, t = 2, 4
	for true {
		n := k*2 + 1
		S := n*3 + 1
		if S > 1e9 {
			break
		}
		if S > 0 {
			ans2 += S
			fmt.Println(n, n, n+1)
		}
		k, t = -2*k-t-1, -3*k-2*t-1
	}
	fmt.Println(ans1 + ans2)
}
